Mathlib/Dynamics/TopologicalEntropy/CoverEntropy.lean

```diff
@@ -0,0 +1,590 @@
+      Prod.map_id, preimage_id_eq, id_eq, mem_iUnion]
+    use y, y_s
+  rcases this.iterate_le_pow F_inv V_symm n with ⟨t, t_dyncover, t_card⟩
+  rw [one_mul n] at t_dyncover
+  exact ⟨t, t_dyncover.of_entourage_subset V_U⟩
+
+/-! ### Minimal cardinal of dynamical covers -/
+
+/-- The smallest cardinal of a `(U, n)`-dynamical cover of `F`. Takes values in `ℕ∞`, and is
+  infinite if and only if `F` admits no finite dynamical cover.-/
+noncomputable def coverMincard (T : X → X) (F : Set X) (U : Set (X × X)) (n : ℕ) : ℕ∞ :=
+  ⨅ (s : Finset X) (_ : IsDynCoverOf T F U n s), (s.card : ℕ∞)
+
+lemma coverMincard_le_card {T : X → X} {F : Set X} {U : Set (X × X)} {n : ℕ} {s : Finset X}
+    (h : IsDynCoverOf T F U n s) :
+    coverMincard T F U n ≤ s.card := iInf₂_le s h
+
+lemma coverMincard_monotone_time (T : X → X) (F : Set X) (U : Set (X × X)) :
+    Monotone (fun n : ℕ ↦ coverMincard T F U n) :=
+  fun _ _ m_n ↦ biInf_mono fun _ h ↦ h.of_le m_n
+
+lemma coverMincard_antitone_entourage (T : X → X) (F : Set X) (n : ℕ) :
+    Antitone (fun U : Set (X × X) ↦ coverMincard T F U n) :=
+  fun _ _ U_V ↦ biInf_mono fun _ h ↦ h.of_entourage_subset U_V
+
+lemma coverMincard_finite_iff (T : X → X) (F : Set X) (U : Set (X × X)) (n : ℕ) :
+    coverMincard T F U n < ⊤ ↔
+    ∃ s : Finset X, IsDynCoverOf T F U n s ∧ s.card = coverMincard T F U n := by
+  refine ⟨fun h_fin ↦ ?_, (fun ⟨s, _, s_coverMincard⟩ ↦ s_coverMincard ▸ WithTop.coe_lt_top s.card)⟩
+  rcases WithTop.ne_top_iff_exists.1 (ne_of_lt h_fin) with ⟨k, k_min⟩
+  rw [← k_min]
+  simp only [ENat.some_eq_coe, Nat.cast_inj]
+  have : Nonempty {s : Finset X // IsDynCoverOf T F U n s} := by
+    by_contra h
+    apply ENat.coe_ne_top k
+    rw [← ENat.some_eq_coe, k_min, coverMincard, iInf₂_eq_top]
+    simp only [ENat.coe_ne_top, imp_false]
+    rw [nonempty_subtype, not_exists] at h
+    exact h
+  have key := ciInf_mem (fun s : {s : Finset X // IsDynCoverOf T F U n s} ↦ (s.val.card : ℕ∞))
+  rw [coverMincard, iInf_subtype'] at k_min
+  rw [← k_min, mem_range, Subtype.exists] at key
+  simp only [ENat.some_eq_coe, Nat.cast_inj, exists_prop] at key
+  exact key
+
+@[simp]
+lemma coverMincard_empty {T : X → X} {U : Set (X × X)} {n : ℕ} : coverMincard T ∅ U n = 0 :=
+  le_antisymm (sInf_le (by simp [IsDynCoverOf])) (zero_le (coverMincard T ∅ U n))
+
+lemma coverMincard_eq_zero_iff (T : X → X) (F : Set X) (U : Set (X × X)) (n : ℕ) :
+    coverMincard T F U n = 0 ↔ F = ∅ := by
+  refine Iff.intro (fun h ↦ subset_empty_iff.1 ?_) (fun F_empt ↦ by rw [F_empt, coverMincard_empty])
+  have := coverMincard_finite_iff T F U n
+  rw [h, eq_true ENat.zero_lt_top, true_iff] at this
+  simp only [IsDynCoverOf, Finset.mem_coe, Nat.cast_eq_zero, Finset.card_eq_zero, exists_eq_right,
+    Finset.not_mem_empty, iUnion_of_empty, iUnion_empty] at this
+  exact this
+
+lemma coverMincard_pos_iff (T : X → X) (F : Set X) (U : Set (X × X)) (n : ℕ) :
+    1 ≤ coverMincard T F U n ↔ F.Nonempty := by
```

Comment:
```suggestion
    0 < coverMincard T F U n ↔ F.Nonempty := by
```
(Oherwise, use `one_le_coverMincard_iff` as the lemma name)

Reply:
Done.

I changed the name of the lemma ; the lower bound `1` is too useful when applying logarithms.